### PR TITLE
wip: dbus activation & single instance support

### PR DESCRIPTION
**Pull request by wash2**
This implements org.freedesktop.Application for cosmic apps.

### CODE REVIEW

**Comment by wash2** [comment]
https://github.com/pop-os/cosmic-settings/pull/95 demonstrates usage for now but I will update the design demo as well.

File: src/process.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+use std::process::{exit, Command, Stdio};
+
+use nix::sys::wait::waitpid;
+use nix::unistd::{fork, ForkResult};
+
+/// Performs a double fork with setsid to spawn and detach a command.
+pub fn spawn(mut command: Command) {
```
[the remark this answers is not in the document]
Comment (wash2) [reply src/process.rs:7]: It is used by applets when they spawn the settings application. I will push my wip on that with usage by the time applet.

**Comment by wash2** [comment]
I guess I can also test this with the launcher and app library. They could probably use this instead of their dbus subscriptions.

**Comment by wash2** [comment]
https://github.com/pop-os/cosmic-launcher/pull/72 has an example of usage with the launcher

**Review by mmstick — APPROVED** [review]

**Pull request merged**